pkg/proc/target_group.go

```diff
@@ -46,6 +55,9 @@ func NewGroup(t *Target) *TargetGroup {
 		LogicalBreakpoints:    t.Breakpoints().Logical,
 		continueOnce:          t.continueOnce,
 	}
+	grp.cctx.grp = grp
```

Comment:
It's not enough, when we add a new process we have to create the target object to apply the breakpoints to it.

Reply:
> It's not enough, when we add a new process we have to create the target object to apply the breakpoints to it.

We can add a new `StopReason` like `ProcessCreated` or something and let `TargetGroup.Continue` handle it accordingly. That would actually prevent the need for maintaining a list of new processes on the context object at all. We still have the problem of keeping `cctx.procs` in sync, but it's relatively easy to just assign it at the top of the loop right before we call `grp.continueOnce`.

Reply:
We'd have to stop every other thread of every process to do that, it's pretty wasteful to exit the ContinueOnce loop if we just inted to re-enter it anyway.

Reply:
That's a good point. Let me think on this a bit more.

Reply:
What if we got rid of `ContinueOnceContext` and moved the members directly into `TargetGroup`? I think this assignment and circular reference shows that `ContinueOnceContext` is redundant. It's a long lived type that exists as long as the `TargetGroup` and is not independent in any way, so why separate it? Then, so we don't have to expose all of TargetGroup into the backends we have them accept an interface exposing only the methods they need instead of taking a `ContinueOnceContext` object.